### ISSUE TITLE
Positive Preserving Scheme

### ISSUE THREAD

**Issue by Zhengyu-Huang** [CliMA/ClimaAtmos.jl#1679]
## Describe the bug
In our simulation, to remain positiveness for the tracer, we have
- vertical upwind scheme
- horizontal limiter (require SSP time stepper)

## To Reproduce
We have the following issues
- We still have negative tracer values
- The maximum dt for SSP time stepper and the vertical upwind scheme is small

Debug strategies
- test other SSP schemes (i.e., SSP332) (see #1691 )
- Single-column upwinding tests using 1D EDMFX test setup (see #1772 )
- try explicit treatment of the tracer variables (passive tracer)
- test positive pressing for horizontal limiter (advection equation) 
     - @dennisYatunin  and @valeriabarra  both tested it and reproduced the original results from the paper. You can see the report [here](https://clima.github.io/ClimaTimeSteppers.jl/dev/dev/report_gen/#Errors-with-a-Limiter). You can see that with high enough horizontal resolution, `h_elem 20`, with SSP333, we reach undershoots that are only of the order of 1.5802e-13
- use exact Jacobian for Zalesak
     - this can be done only after the Implicit Solver interface update is complete (CC SDI https://github.com/CliMA/ClimaCore.jl/issues/1230)

**Comment by valeriabarra** [comment]
For the second bullet point: test single-column configuration, I tested the single-column moist configuration with ARS time-stepping scheme, and different upwinding modes:

- 1st-order upwinding: 
CLI options used: 
```
julia -i --color=yes --project=examples examples/hybrid/driver.jl --moist equil --hyperdiff false --config column --initial_condition IsothermalProfile --z_max 30e3 --z_elem 45 --dz_bottom 30 --dz_top 5000 --t_end 10days --dt 3hours --dt_save_to_sol 3hours --dt_save_to_disk 5days --job_id tracer_energy_first_single_column_hydrostatic_balance_ft64 --FLOAT_TYPE Float64 --enable_threading false --tracer_upwinding first_order --energy_upwinding first_order
```
extrema: 
```
println(ρq_tot |> parent |> extrema)
(-1.742503024692422e-10, 0.021005971941460477)
```

- centered-diff: 
CLI options used:
```
julia -i --color=yes --project=examples examples/hybrid/driver.jl --moist equil --hyperdiff false --config column --initial_condition IsothermalProfile --z_max 30e3 --z_elem 45 --dz_bottom 30 --dz_top 5000 --t_end 10days --dt 3hours --dt_save_to_sol 3hours --dt_save_to_disk 5days --job_id single_column_hydrostatic_balance_ft64 --FLOAT_TYPE Float64 --enable_threading false
```
extrema:
```
println(ρq_tot |> parent |> extrema)
(-9.322511858552337e-12, 0.020867464918757853)
```

- 3rd-order upwinding:
CLI options used:
```
julia -i --color=yes --project=examples examples/hybrid/driver.jl --moist equil --hyperdiff false --config column --initial_condition IsothermalProfile --z_max 30e3 --z_elem 45 --dz_bottom 30 --dz_top 5000 --t_end 10days --dt 3hours --dt_save_to_sol 3hours --dt_save_to_disk 5days --job_id tracer_energy_third_order_single_column_hydrostatic_balance_ft64 --FLOAT_TYPE Float64 --enable_threading false --tracer_upwinding third_order --energy_upwinding third_order
```
extrema:
```
println(ρq_tot |> parent |> extrema)
(-4.326136769799144e-9, 0.0210060110699763)
```

- Zalesak:
CLI options used:
```
julia -i --color=yes --project=examples examples/hybrid/driver.jl --moist equil --hyperdiff false --config column --initial_condition IsothermalProfile --z_max 30e3 --z_elem 45 --dz_bottom 30 --dz_top 5000 --t_end 10days --dt 3hours --dt_save_to_sol 3hours --dt_save_to_disk 5days --job_id tracer_energy_zalesak_upwind_single_column_hydrostatic_balance_ft64 --FLOAT_TYPE Float64 --enable_threading false --tracer_upwinding zalesak --energy_upwinding zalesak
```
extrema:
```
println(ρq_tot |> parent |> extrema)
(-7.895641000204533e-9, 0.021006533151996198)
```

At the moment, I'm trying to re-test all of these also with SSP time-stepping but for some reason I can't get it to work in the single-column configuration, even with very small `dt`s. I'll keep looking into this.

**Comment by Zhengyu-Huang** [comment]
> 

It is weird that 1) we still have negative value for first order upwind scheme 2) central difference is better than first order upwind scheme

**Comment by szy21** [comment]
Superseded by ClimaCore Issue: https://github.com/CliMA/ClimaCore.jl/issues/1327
